compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -2276,6 +2278,10 @@ genLoadLongConstant(TR::CodeGenerator * cg, TR::Node * node, int64_t value, TR::
       TR::Instruction * temp = cursor;
       cursor = generateRegLitRefInstruction(cg, TR::InstOpCode::LG, node, targetRegister, value, TR_BodyInfoAddress, cond, cursor, base);
       }
+   else if (cg->comp()->compileRelocatableCode() && sym && sym->isCatchBlockCounter())
```

Comment:
It's better to replace `compileRelocatableCode()` with something like `needRelocationsForBodyInfoData()`